includes/managers/controls.php

```diff
@@ -1347,6 +1362,29 @@ private function promotion_switcher_control( $title, $id ): string {
 		</div>';
 	}
 
+	private function react_promotion_control( $title, $id ): string {
+		return '<div data-promotion="' . $id . '" class="elementor-control-type-switcher elementor-label-inline e-control-motion-effects-promotion__wrapper">
+			<div class="elementor-control-content">
+				<div class="elementor-control-field">
+					<label>
+						' . $title . '
+					</label>
+					<span class="e-control-motion-effects-promotion__lock-wrapper">
+						<i class="eicon-lock"></i>
+					</span>
+					<div class="elementor-control-input-wrapper">
+						<label class="elementor-switch elementor-control-unit-2 e-control-' . $id . '-promotion">
+							<input type="checkbox" class="elementor-switch-input" disabled>
+							<span class="elementor-switch-label" data-off="Off"></span>
+							<span class="elementor-switch-handle"></span>
+						</label>
+					</div>
+					<div class="e-promotion-react" data-promotion="' . $id . '"></div>
```

Comment:
@hein-obox  I brought this back for now, we can remove this later, to reduce duplication, but I liked the idea that I can get to it from the `rootElement` and not search for "closest", up do debate....

Reply:
@mike-elementor No problem. I understand. The most important aspect that we can filter the correct data immediately on control level.